src/sql/magic.py

```diff
@@ -144,6 +144,20 @@ def __init__(self, shell):
         # Add ourself to the list of module configurable via %config
         self.shell.configurables.append(self)
 
+    @validate("displaylimit")
+    def _valid_displaylimit(self, proposal):
+        value = int(proposal["value"])
```

Comment:
Seems we can't catch the non Integer type, since we already define the type and it got caught by `IPython.core.magic_arguments` earlier, which throws `The 'displaylimit' trait of a SqlMagic instance expected an int...`

Reply:
But I've added the test case